src/ostree/ot-builtin-refs.c

```diff
@@ -60,7 +70,60 @@ static gboolean do_ref (OstreeRepo *repo, const char *refspec_prefix, GCancellab
           g_print ("%s\n", ref);
         }
     }
+  else if(opt_create)
+    {
+      gboolean old_ref_exists = FALSE;
+      const char *checksum = NULL;
+
+      g_hash_table_iter_init (&hashiter, refs);
+      while (g_hash_table_iter_next (&hashiter, &hashkey, &hashvalue))
+        {
+          g_autofree char *ref = NULL;
+          g_autofree char *remote = NULL;
+          const char *existing_ref= hashkey;
+
+          if (!ostree_parse_refspec (refspec_prefix, &remote, &ref, error))
+            goto out;
+
+          if(strcmp (opt_create, existing_ref) == 0)
+            {
+              g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                         "--create specified but ref %s already exists", existing_ref);
+              goto out;
+            }
+
+          if(strcmp(refspec_prefix, hashkey) == 0)
+            {
+              old_ref_exists = TRUE;
+              checksum = hashvalue;
+            }
+        }
+
+      if(old_ref_exists)
+        {
+          if (!ostree_repo_prepare_transaction (repo, NULL, cancellable, error))
+            goto out;
+
+          ostree_repo_transaction_set_ref (repo, NULL, opt_create, checksum);
+
+          if (!ostree_repo_resolve_rev (repo, opt_create, TRUE, &parent, error))
```

Comment:
Ahhh I see, I thought it was only `set_ref_immediate` that suffered from the tmp issue, but reading https://github.com/ostreedev/ostree/issues/338 says otherwise. Once that issue is fixed though, all this should probably be dropped in favour of a single `set_ref_immediate` call.

Also, shouldn't `allow_noent` be `FALSE` here?